src/pudl/metadata/resources/eiaaeo.py

```diff
@@ -0,0 +1,514 @@
+"""Table definitions for EIA AEO resources."""
+
+from typing import Any
+
+# 2024-04-24: workaround so we can *define* a bunch of schemas without
+# documenting them or expecting assets to exist
+_STAGING_RESOURCE_METADATA: dict[str, dict[str, Any]] = {
+    "core_eiaaeo__generation_electric_sector": {
+        "description": (
+            "Projected generation capacity & total generation in the electric "
+            "sector."
+        ),
+        "schema": {
+            "fields": [
+                "report_year",
+                "electricity_market_module_region",
+                "aeo_case",
+                "projection_date",
+                "aeo_electricity_sector_fuel_type",
+                "net_summer_capacity_mw",  # should this be capacity_mw_eia?
+                # should we include "first modeled year" as a field, to help interpret all the cumulative fields?
+                "net_summer_capacity_cumulative_planned_additions_mw",
+                "net_summer_capacity_cumulative_unplanned_additions_mw",
+                "net_summer_capacity_cumulative_total_additions_mw",  # this is just planned + unplanned - do we even care about publishing this?
+                "net_summer_capacity_cumulative_retirements_mw",
+                "gross_generation_mwh",  # chose gross bc the numbers add up to "total generation" which then equals "sales to customers + generation for own use" - but... is that right?
```

Comment:
i.e., the "renewable" + "coal" + "whatever" + ... all add up to the "total generation" projected for the year. Which then is equal to "sales + generation for own use" - which sounds to me like gross, not net, generation.